pkg/apis/pipeline/v1alpha1/task_validation.go

```diff
@@ -88,6 +93,37 @@ func (ts *TaskSpec) Validate() *apis.FieldError {
 	return nil
 }
 
+func validateVolumes(volumes []corev1.Volume) *apis.FieldError {
+	// Task must not have duplicate volume names.
+	vols := map[string]struct{}{}
+	for _, v := range volumes {
+		if _, ok := vols[v.Name]; ok {
+			return apis.ErrMultipleOneOf("name")
+		}
+		vols[v.Name] = struct{}{}
+	}
+	return nil
+}
+
+func validateSteps(steps []corev1.Container) *apis.FieldError {
+	// Task must not have duplicate step names.
+	names := map[string]struct{}{}
+	for _, s := range steps {
+		if s.Image == "" {
+			return apis.ErrMissingField("Image")
+		}
+
+		if s.Name == "" {
+			continue
+		}
+		if _, ok := names[s.Name]; ok {
+			return apis.ErrMultipleOneOf("name")
```

Comment:
I think `ErrMultipleOneOf` is meant to signal that multiple mutually-exclusive fields are specified at the same time. Like a `TaskRun` specifying a `TaskSpec` and a `TaskRef`.

I'm not sure whether there's a field error we should use here, we can't be the only ones who want to prohibit multiple identical keys in a map-like structure. [`ErrInvalidValue`](https://godoc.org/github.com/knative/pkg/apis#ErrInvalidValue) might be the best bet?

Reply:
oh right, that's what was used previously, but I do agree :+1: